### PR TITLE
[Sol->Yul] Skipping mapping types when deleting structs and arrays.

### PR DESCRIPTION
**Pull request by mijovic**
Part of #10390

### CODE REVIEW

File: libsolidity/codegen/YulUtilFunctions.cpp [hunk 1]
```diff
@@ -1023,35 +1023,42 @@ std::string YulUtilFunctions::resizeArrayFunction(ArrayType const& _type)
...
+					}
+				</needsClearing>
+			})");
+			templ("functionName", functionName);
+			templ("maxArrayLength", (u256(1) << 64).str());
+			templ("panic", panicFunction());
+			templ("fetchLength", arrayLengthFunction(_type));
+			templ("isDynamic", _type.isDynamicallySized());
+			bool isMappingBase = _type.baseType()->category() == Type::Category::Mapping;
+			templ("needsClearing", !isMappingBase);
```
Comment (leonardoalt) [inline libsolidity/codegen/YulUtilFunctions.cpp:1148]: Could just inline it here

Comment (leonardoalt) [reply libsolidity/codegen/YulUtilFunctions.cpp:1148]: Ah no, also used below

Comment (leonardoalt) [reply libsolidity/codegen/YulUtilFunctions.cpp:1148]: Weird, I thought this was somehow done before, as part of the `zeroValueFunction` or something like that

Comment (mijovic) [reply libsolidity/codegen/YulUtilFunctions.cpp:1148]: Hm, there is an assertion in zeroValueFunction.
I wanted to do it this way as it was doing the same in old codegen. But, other option is to implement noop zero function for `MappingType` which would be maybe weird...

**Comment by mijovic** [comment]
Rebased

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**